### PR TITLE
[MRESOLVER-130] Move GlobalSyncContextFactory to a separate module

### PR DESCRIPTION
**Pull request by michael-o**
@elharo Please check documentation for typos and consistency.

### CODE REVIEW

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultSyncContextFactory.java [hunk 1]
```diff
@@ -30,59 +28,31 @@
...
 import org.eclipse.aether.metadata.Metadata;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import com.google.inject.Singleton;
 
 /**
- * A factory to create synchronization contexts. This default implementation uses fair global locking
- * based on {@link ReentrantReadWriteLock}. Explicit artifacts and metadata passed are ignored.
+ * A factory to create synchronization contexts. This default implementation actually does not provide any real
```
Comment (elharo) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultSyncContextFactory.java:33]: delete "actually"

File: maven-resolver-synccontext-global/pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <artifactId>maven-resolver-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-impl</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
```
Comment (elharo) [inline maven-resolver-synccontext-global/pom.xml:56]: Where is this provided from? Or are you using this as compile only?

File: maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ * Manages access to a properties file. This override drops internal synchronization becauses it
+ * relies on external synchronization provided by outer {@link SyncContext} instances.
+ */
+class TrackingFileManager
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger( TrackingFileManager.class );
+
+    public Properties read( File file )
```
Comment (elharo) [inline maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:46]: no need for public methods in non-public class

File: maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        FileInputStream stream = null;
+        try
```
Comment (elharo) [inline maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:49]: I think you can use try with resources here

File: maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        Properties props = new Properties();
+
+        File directory = file.getParentFile();
+        if ( !directory.mkdirs() && !directory.exists() )
+        {
+            LOGGER.warn( "Failed to create parent directories for tracking file {}", file );
+            return props;
+        }
+
+        RandomAccessFile raf = null;
```
Comment (elharo) [inline maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:86]: try with resources?

File: maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        try
+        {
+            raf = new RandomAccessFile( file, "rw" );
+
+            if ( file.canRead() )
+            {
+                byte[] buffer = new byte[(int) raf.length()];
+
+                raf.readFully( buffer );
```
Comment (elharo) [inline maven-resolver-synccontext-global/src/main/java/org/eclipse/aether/internal/impl/TrackingFileManager.java:95]: Files.readAllBytes?

File: maven-resolver-synccontext-global/pom.xml (hunk 2 above)
Comment (michael-o) [reply maven-resolver-synccontext-global/pom.xml:56]: I have copied this block from `maven-resolver-impl`. The dependency is delivered with Maven. I'd like to leave it as-is or modify as a whole which is out of scope here.

**Comment by michael-o** [comment]
@elharo Regarding comments to `TrackingFileManager`. I have intentionally not touched the class as a whole because I have copied it from `maven-resolver-impl`. I needed to remove the synchronization only because I consider it to be wrong for several reasons. A JIRA issue will follow. So let's also keep this class as-is (out of scope).

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultSyncContextFactory.java (hunk 1 above)
Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultSyncContextFactory.java]: Done!

**Comment by jira-importer** [comment]
Resolve #917

**Pull request closed without merging**